transcom/mymove
# Use new admin users table to authentication and authorization.

## Description

We currently [can't log into](https://docs.google.com/document/d/1Sb6S5bgsqyvfyxJyqLusD5s3n4ZmNDSjdPkEeMrCFYI/edit#) the admin app on dev, experimental, or staging. We're also [not properly authorizing users](https://www.pivotaltracker.com/n/projects/2136866/stories/166656598) for hitting admin app endpoints. The changes in this PR take advantage of the new `admin_users` table to make use of a strategy very similar to what we use to authenticate office users, both unknown (logging in for the first time) and known.

## Reviewer Notes

- How do we feel about the `AdminAuthMiddleware`?
- I've created a secure migration for bootstrapping us with a few system admin users:
  - Risa (the new product person on this feature)
  - Alexi (the original product person on this feature)
  - Me
  - Shall I go ahead and add more? The purpose of this is so we can begin verifying admin stories in staging.

## Setup

- In `local_migrations/20190710174257_add_system_admin_users.sql`, edit one of the insert statements to reflect the email address you use with login.gov.
- `make db_dev_e2e_migrate`
- `make server_run`
- `make client_run`

## Code Review Verification Steps

- Visit `adminlocal:3000`
- Log in via login.gov with the email address you entered in the local migration
- Ensure you don't see an unauthorized page
- Log out
- Login in as an office user, either devlocal or via login.gov
- Attempt to visit `officelocal:3000/admin/v1/office_users` and ensure you hit an `Unauthorized` page



## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/167018198) for this change
